src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx

```diff
@@ -375,22 +375,22 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
 
   const handleThorchainSaversEmptyClick = useCallback(() => setHideEmptyState(true), [])
 
-  if (!earnOpportunityData || isTradingActiveLoading) {
+  if (isTradingActiveLoading) {
     return (
       <Center minW='500px' minH='350px'>
         <CircularProgress />
       </Center>
     )
   }
 
-  if (!(maybeAccountId && opportunityDataFilter)) return null
-  if (!asset) return null
-  if (!underlyingAssetsWithBalancesAndIcons || !earnOpportunityData) return null
-
   if (bnOrZero(underlyingAssetsFiatBalanceCryptoPrecision).eq(0) && !hideEmptyState) {
     return <ThorchainSaversEmpty assetId={assetId} onClick={handleThorchainSaversEmptyClick} />
   }
 
+  if (!(maybeAccountId && opportunityDataFilter)) return null
+  if (!asset) return null
+  if (!underlyingAssetsWithBalancesAndIcons || !earnOpportunityData) return null
```

Comment:
Not sure about this one - this was previously derp and would never get hit because of l.378, but now will.

Will this now render an empty state as things are loading?

Reply:
Confirmed this will produce a regression - things are turbo fast on a fast connection but slow 3g makes it obvious
https://jam.dev/c/cc18b779-a2ab-4b7f-ae6d-718776f560b3

Reply:
If we use `!earnOpportunityData` to display the loader, it will load infinitely because `maybeAccountId` will never be defined and so `earnOpportunityData`, we need to check if `maybeAccountId` is defined BUT `earnOpportunityData` is actually not defined so the loader won't be shown if `maybeAccountId` and `earnOpportunityData` is not defined

Reply:
Still able to repro after https://github.com/shapeshift/web/pull/7082/commits/03a30e9183abf7b64e2303f907583bfb5a388df3 @NeOMakinG but confirmed this isn't a regression - PR incoming on top of this